lusidtools/cocoon/seed_sample_data.py

```diff
@@ -0,0 +1,69 @@
+import pandas as pd
+from lusidtools.cocoon.cocoon import load_from_data_frame
+from lusidtools.cocoon.utilities import load_json_file
+import logging
+
+logger = logging.getLogger()
+
+mappings = dict(load_json_file("config/seed_sample_data.json"))
+
+
+def seed_data(
+    api_factory,
+    domains,
+    scope: str,
+    transaction_source,
+    mappings: dict = mappings,
+    sub_holding_keys=[],
+):
+
+    """
+    Description:
+    This function allows users to seed their LUSID environment with some core data (e.g. instruments,
+    portfolios, and transaction) from one file.
+
+    :param lusid.utilities.ApiClientFactory api_factory: The api factory to use.
+    :param list[str] domains: A list of the file_types for upload.
+    :param str scope: The scope of the transaction portfolio.
+    :param str transaction_source: a CSV or XLSX file of data.
+    :param dict mappings: a file containing mapping of CSV or DataFrame headers to LUSID headers.
+    """
+
+    if str(transaction_source).endswith(".csv"):
```

Comment:
this won't work with `.CSV`, `.Csv`, `.txt`

I think its better to have explicit parameter which informs what file type to use, we can then easily make it support xml later

```
...
transaction_source,
file_type,
```
...
remember to do a case insensitive comparison when checking what is passed in

Reply:
Sure, thats done.
(1) New parameter added to logic
(2) Added case insensitive check